packages/@aws-cdk/core/lib/assets.ts

```diff
@@ -238,9 +238,10 @@ export interface FileAssetLocation {
 
   /**
    * The HTTP URL of this asset on Amazon S3.
+   * @default - value specified in `httpUrl` is used.
    * @deprecated use `httpUrl`
    */
-  readonly s3Url: string;
+  readonly s3Url?: string;
```

Comment:
This is technically a breaking change because `FileAssetLocation` is an output struct, but since it's not really used by anyone but the framework, I think it's fine.